.github/workflows/benchmark.yaml

```diff
@@ -0,0 +1,295 @@
+      - name: Start HTTP Server to Expose Metrics
+        run: |
+          cat << 'EOF' > server.py
+          from http.server import SimpleHTTPRequestHandler, HTTPServer
+
+          class MetricsHandler(SimpleHTTPRequestHandler):
+              def do_GET(self):
+                  if self.path == "/metrics":
+                      self.send_response(200)
+                      self.send_header("Content-type", "text/plain")
+                      self.end_headers()
+                      with open("metrics.txt", "r") as f:
+                          self.wfile.write(f.read().encode())
+                  else:
+                      self.send_response(404)
+                      self.end_headers()
+
+          if __name__ == "__main__":
+              server = HTTPServer(('0.0.0.0', 9000), MetricsHandler)
+              print("Serving on port 9000...")
+              server.serve_forever()
+          EOF
+          
+          nohup python3 server.py &
+
+      - name: Wait for Prometheus to Collect Data
+        run: sleep 30
+
+      - name: Check Prometheus targets page
+        run: |
+          http_status=$(curl -o /dev/null -s -w "%{http_code}" http://localhost:9090/targets)
+          if [ "$http_status" -eq 200 ]; then
+            echo "Prometheus targets page is reachable."
+          else
+            echo "Error: Prometheus targets page is not reachable. Status code: $http_status"
+            exit 1
+          fi
+
+          http_status=$(curl -o /dev/null -s -w "%{http_code}" http://localhost:9090/targets)
+          if [ "$http_status" -eq 200 ]; then
+            echo "Prometheus targets page is reachable."
+            
+            # Check for lastError field in the targets API
+            error=$(curl -s http://localhost:9090/api/v1/targets | jq -r '.data.activeTargets[].lastError')
+            if [ "$error" != "null" ] && [ -n "$error" ]; then
+              echo "Error: Prometheus target has an error: $error"
+              exit 1
+            else
+              echo "No errors found in Prometheus targets."
+            fi
+
+          else
+            echo "Error: Prometheus targets page is not reachable. Status code: $http_status"
+            exit 1
+          fi
+
+      # - name: Debug via SSH
+      #   uses: mxschmitt/action-tmate@v3
+
+      - name: Check Benchmark Metrics Against Threshold
+        run: |
+          MAX_TIME_NS=1200000000  # 1.2s
+          MAX_ALLOCS=4000
+          MAX_MEM_BYTES=450000
+
+          # Query Prometheus Metrics, get the max value
+          time_ns=$(curl -s "http://localhost:9090/api/v1/query?query=max(benchmark_createclustercatalog_ns)" | jq -r '.data.result[0].value[1]')
+          allocs=$(curl -s "http://localhost:9090/api/v1/query?query=max(benchmark_createclustercatalog_allocs)" | jq -r '.data.result[0].value[1]')
+          mem_bytes=$(curl -s "http://localhost:9090/api/v1/query?query=max(benchmark_createclustercatalog_mem_bytes)" | jq -r '.data.result[0].value[1]')
```

Comment:
From controller-runtime we have those metrics which seem relevant for performance, see:

- Reconciliation Time → controller_runtime_reconcile_time_seconds_sum
- HTTP Request Duration → catalogd_http_request_duration_seconds_sum
- Number of Reconciliations → controller_runtime_reconcile_total
- CPU Usage → go_cpu_classes_gc_total_cpu_seconds_total
- Memory Usage → go_cpu_classes_idle_cpu_seconds_total
- Failed reconciliations → controller_runtime_reconcile_errors_total
- Webhook failures → controller_runtime_webhook_latency_seconds